language_service/src/identifier_locator.rs

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::mem::replace;
+use std::rc::Rc;
+
+use crate::qsc_utils::{find_ident, find_item, span_contains, span_touches, Compilation};
+use qsc::ast::visit::{walk_expr, walk_namespace, walk_pat, walk_ty_def, Visitor};
+use qsc::{ast, hir, resolve};
+
+#[allow(unused_variables)]
+pub(crate) trait LocatorAPI<'package> {
+    fn at_callable_def(
+        &mut self,
+        context: &LocatorContext<'package>,
+        decl: &'package ast::CallableDecl,
+    ) {
+    }
+
+    fn at_callable_ref(
+        &mut self,
+        path: &'package ast::Path,
+        item_id: &'package hir::ItemId,
+        item: &'package hir::Item,
+        package: &'package hir::Package,
+        decl: &'package hir::CallableDecl,
+    ) {
+    }
+
+    fn at_new_type_def(&mut self, type_name: &'package ast::Ident, def: &'package ast::TyDef) {}
+
+    fn at_new_type_ref(
+        &mut self,
+        path: &'package ast::Path,
+        item_id: &'package hir::ItemId,
+        item: &'package hir::Item,
+        package: &'package hir::Package,
+        type_name: &'package hir::Ident,
+        udt: &'package hir::ty::Udt,
+    ) {
+    }
+
+    fn at_field_def(
+        &mut self,
+        context: &LocatorContext<'package>,
+        field_name: &'package ast::Ident,
+        ty: &'package ast::Ty,
+    ) {
+    }
+
+    fn at_field_ref(
+        &mut self,
+        expr_id: &'package ast::NodeId,
+        field_ref: &'package ast::Ident,
+        item_id: &'package hir::ItemId,
+        field_def: &'package hir::ty::UdtField,
+    ) {
```

Comment:
In the spirit of keeping the "the first parameter is the thing that the cursor is on", I suggest putting the `field_ref` parameter at the top. All the other `*_ref` methods already follow this pattern